0.data_analysis/nbconverted/plates_1_2_sig_test_nf1_sc_all_cellprofiler.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Obnibus and post hoc testing with Anova and scheffe's test
+
+# ## Imports
+
+# In[1]:
+
+
+import sys
+from sklearn.ensemble import IsolationForest
+import scikit_posthocs as sp
+from pathlib import Path
+import numpy as np
+import pandas as pd
+from collections import defaultdict
+
+sys.path.append("../utils")
+import analysis_utils as au
+import preprocess_utils as ppu
+
+
+# In[2]:
+
+
+rnd_val = 0 # Random value for all seeds
+rng = np.random.default_rng(seed=rnd_val) # random number generator
+
+
+# # Preprocess data using preprocess utils
+
+# In[3]:
+
+
+current_dir = Path.cwd()
+print(current_dir)
+
+
+# In[4]:
+
+
+filename = 'nf1_sc_all_cellprofiler.csv.gz'
+base_path = Path("../nf1_benchmark_repo/4_processing_features/data")
+ending_path = Path("CellProfiler/nf1_sc_all_cellprofiler.csv.gz")
+
+plates = {}
+
+plates['1'], plates['2'] = {"path": base_path / "Plate1" / ending_path}, {"path": base_path / "Plate2" / ending_path}
+
+for plate, vals in plates.items():
+    # Initializes the preprocessing class
+    po = ppu.Preprocess_data(path=vals['path'])
+    
+    # Gets the dataframe after removing metadata columns, 
+    # except for the 'Metadata_genotype' column specified above
+    platedf = po.remove_meta(po.df, kept_meta_columns=['Metadata_genotype'])
+    platedf.drop(['Cytoplasm_Number_Object_Number'], axis=1, inplace=True)
+    
+    # Use isolation forest to select inliers:
+    isof = IsolationForest(random_state = rnd_val) 
+    out_preds = isof.fit_predict(platedf.drop(columns=['Metadata_genotype']))
+    ind = np.nonzero(out_preds == 1)[0] # Select inliers
+    
+    # Select inlier samples:
+    plates[plate]['df'] = platedf.iloc[ind]
+    
+platesdf = [plates[key]['df'] for key in plates]
+
+
+# # Conduct testing with scheffe's test
+
+# In[5]:
+
+
+#gtypes = ['Null','WT'] # The 2 types of genotypes
```

Comment:
Recommend removing commented-out code lines (this one and the one below seem to be commented out and not used).

Reply:
Changed